README.md

```diff
@@ -123,5 +123,146 @@ LabelCell(key: "company", state: LabelState(text: "Shopify",
+backgroundColor = .white
+setupConstraints()
+}
+
+required init?(coder aDecoder: NSCoder) {
+fatalError("init(coder:) has not been implemented")
+}
+
+func setupConstraints() {
+profilePicture.translatesAutoresizingMaskIntoConstraints = false
+addSubview(profilePicture)
+
+addSubview(nameLabel)
+addSubview(messageLabel)
+
+timeStampLabel.translatesAutoresizingMaskIntoConstraints = false
+addSubview(timeStampLabel)
+
+let stackView = UIStackView(arrangedSubviews: [nameLabel, messageLabel])
+stackView.axis = .vertical
+stackView.alignment = .fill
+stackView.spacing = 3
+stackView.distribution = .fill
+stackView.translatesAutoresizingMaskIntoConstraints = false
+addSubview(stackView)
+
+let constraints: [NSLayoutConstraint] = [
+profilePicture.heightAnchor.constraint(equalToConstant: 60),
+profilePicture.widthAnchor.constraint(equalToConstant: 60),
+profilePicture.leadingAnchor.constraint(equalTo: leadingAnchor),
+profilePicture.topAnchor.constraint(equalTo: topAnchor, constant: 12),
+profilePicture.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -12),
+stackView.centerYAnchor.constraint(equalTo: profilePicture.centerYAnchor),
+stackView.topAnchor.constraint(greaterThanOrEqualTo: topAnchor),
+stackView.bottomAnchor.constraint(lessThanOrEqualTo: bottomAnchor),
+stackView.leadingAnchor.constraint(equalTo: profilePicture.trailingAnchor, constant: 12),
+stackView.trailingAnchor.constraint(lessThanOrEqualTo: timeStampLabel.leadingAnchor, constant: -12),
+timeStampLabel.trailingAnchor.constraint(equalTo: trailingAnchor),
+timeStampLabel.topAnchor.constraint(equalTo: topAnchor)
+]
+NSLayoutConstraint.activate(constraints)
+
+profilePicture.layer.borderWidth = 1
+profilePicture.layer.borderColor = UIColor.black.cgColor
+profilePicture.layer.cornerRadius = 30
+profilePicture.backgroundColor = UIColor.gray
+
+nameLabel.text = "FirstName LastName"
+nameLabel.font = UIFont.systemFont(ofSize: 18, weight: .medium)
+messageLabel.text = "This is a new message"
+messageLabel.font = UIFont.systemFont(ofSize: 15, weight: .light)
+
+timeStampLabel.font = UIFont.systemFont(ofSize: 12, weight: .light)
+timeStampLabel.text = "Date Label"
+timeStampLabel.setContentHuggingPriority(.required, for: .horizontal)
+}
+}
+```
+
+Next, we'll build the `MessageViewState` struct that will take care of setting all the views properties:
+
+```
+struct Message {
+let name: String
+let image: UIImage
+let date: Date
+let text: String
+}
+
+struct MessageViewState: Equatable {
+
+let date: Date
+let name: String
+let message: String
+let image: UIImage
+
+init(message: Message) {
+self.date = message.date
+self.name = message.name
+self.message = message.text
+self.image = message.image
+}
+
+static func updateView(_ view: MessageView, state: MessageViewState?) {
+guard let state = state else { return }
+view.timeStampLabel.text = "\(state.date)"
+view.nameLabel.text = state.name
+view.messageLabel.text = state.message
+view.profilePicture.image = state.image
+}
+
```

Comment:
All this code needs formatting. For simplicity (since this is getting pretty long) Why not use directly MessageViewState. instead of wrapping everything into `struct Message`

Reply:
I had it with the multiple values at first but thought it would be simpler to explain the relationship between the model and state objects by passing it directly